cfme/tests/configure/test_proxy.py

```diff
@@ -25,22 +28,43 @@ def proxy_machine():
     This fixture uses for deploy vm on provider from yaml and then receive it's ip
     After test run vm deletes from provider
     """
-    depot_machine_name = random_vm_name('proxy')
-    data = conf.cfme_data.get("proxy_template")
-    if data is None:
-        pytest.skip('Missing proxy_template in cfme_data.yaml, cannot deploy proxy')
-    proxy_provider_key = data["provider"]
-    proxy_template_name = data["template_name"]
-    proxy_port = data['port']
-    vm = deploy_template(proxy_provider_key,
-                         depot_machine_name,
-                         template_name=proxy_template_name)
-    wait_for(func=lambda: vm.ip is not None,
-             num_sec=300,
-             delay=10,
-             message='Waiting for instance "{}" ip to be present.'.format(vm.name))
-
-    yield vm.ip, proxy_port
+    try:
+        data = cfme_data.proxy_template
+        proxy_port = data.port
+        vm = deploy_template(
+            data.provider,
+            random_vm_name('proxy'),
+            template_name=data.template_name
+        )
+    except AttributeError:
+        msg = 'Missing data in cfme_data.yaml, cannot deploy proxy'
+        logger.exception(msg)
+        pytest.skip(msg)
+
+    def get_pingable(vm_mgmt):
```

Comment:
I don't like that I'm repeating this block, because most times its consistently used, but I inserted it into places where the wrapanapi VM object is used directly, instead of a VM object from int_tests. This way it can be easily modified for specific tests/fixtures needs.

Reply:
Took my own advice and created function in `cfme.utils.net`